### PR TITLE
[cleanup] FFmpeg deprecated API calls

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Replace deprecated API calls with no deprecated ones.

## Motivation and Context
silence the warnings
@fritsch I picked these commits from a closed PR you created

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fritsch** [comment]
Wait. That works better. No need to replace old API calls with just new ones. Let's better use the new API as intended. That's why I did not push on my original work.

File: xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp [hunk 1]
```diff
@@ -285,7 +285,13 @@ int CAEEncoderFFmpeg::Encode(uint8_t *in, int in_size, uint8_t *out, int out_siz
   m_Pkt.data = out;
 
   /* encode it */
-  int ret = avcodec_encode_audio2(m_CodecCtx, &m_Pkt, frame, &got_output);
+  int ret = avcodec_send_frame(m_CodecCtx, frame);
+
+  if (ret >= 0)
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp]: why do you test for >= 0? 0: is success; > 0: is undefined.

**Comment by candrews** [comment]
*bump* Any news on this one?

**Comment by fritsch** [comment]
Nothing to bump - feel free to properly use the  new API as it is intended. When I wrapped the old behaviour into the new API calls I realized that it makes no sense.

PR welcome.

**Pull request closed without merging**